### PR TITLE
Tweaks the Light Replacer

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
The Light Replacer now eats glass sheets and lightbulbs when the user clicks on them rather than having to pick up the bulb/sheet and then insert it. This is how shards currently behave and brings them in line.

It now tells you how many fragments you have when you insert any kind of compatible item.

Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).

## Why It's Good For The Game
It's more convenient for human janitors.

Janiborgs are now able to actually use lightbulbs and sheets.

Conservation of mass has been achieved.

## Testing
I used my light replacer on shards and glass and lights and they worked.

## Changelog
:cl:
tweak:  Light replacers can now eat glass and lightbulbs directly to refill.
tweak:  Shards will restore an entire lightbulb when eaten by the light replacer.
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
now that jani borgs can refill their replacers with glass, you should remove the free lights from charges :)

**Comment by DGamerL** [comment]
> Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).

That costs welding fuel though...

> now that jani borgs can refill their replacers with glass, you should remove the free lights from charges :)

Seconded

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -88,25 +88,26 @@
 			return
 		if(!user.unEquip(I))
 			return
-		AddUses(round(increment * 0.75))
+		AddUses(round(increment * 1)) // It literally contains the same amount of glass as a sheet.
```
Comment (DGamerL) [inline code/game/objects/items/devices/lightreplacer.dm:91]:
```suggestion
		AddUses(increment)
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 2]
```diff
@@ -87,8 +93,7 @@ GLOBAL_LIST_INIT(glass_recipes, list (
 		G.use(1)
 		if(!G && !RG && replace)
 			user.put_in_hands(RG)
-	else
-		return ..()
```
Comment (DGamerL) [inline code/game/objects/items/stacks/sheets/glass.dm:91]: Why remove the parent call here?

File: code/game/objects/items/devices/lightreplacer.dm [hunk 3]
```diff
@@ -88,25 +88,26 @@
...
 			if(!user.unEquip(L))
 				return
-			to_chat(user, "<span class='notice'>You insert [L] into [src].</span>")
-			AddShards(1, user)
+			AddUses(1)
+			to_chat(user, "<span class='notice'>You insert [L] into [src]. You have [uses] light\s remaining.</span>")
 			qdel(L)
+			return
+
+		if(!user.unEquip(L))
```
Comment (DGamerL) [inline code/game/objects/items/devices/lightreplacer.dm:106]: You refactored the `if`-statement to pass to this `if(!user.unEquip(..` if it has max uses. Will this add shards over the cap?

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 4]
```diff
@@ -67,6 +67,10 @@ GLOBAL_LIST_INIT(glass_recipes, list (
 
 /obj/item/stack/sheet/glass/attackby(obj/item/W, mob/user, params)
 	..()
+	if(istype(W, /obj/item/lightreplacer))
+		W.attackby(src, user)
+		return
+
```
Comment (DGamerL) [inline code/game/objects/items/stacks/sheets/glass.dm:73]: Does this have to be here? Can't you put this in the `attack_obj()` of the light replacer?

File: code/modules/power/lights.dm [hunk 5]
```diff
@@ -947,8 +946,12 @@
 
 
 // attack bulb/tube with object
-// if a syringe, can inject plasma to make it explode
+// if a syringe, can inject plasma to make it explode. Light replacers eat them.
 /obj/item/light/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/lightreplacer))
+		I.attackby(src, user)
+		return
```
Comment (DGamerL) [inline code/modules/power/lights.dm:953]: Ditto here

**Review by DGamerL — CHANGES_REQUESTED** [review]

**Comment by 1080pCat** [comment]
> > Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).
> 
> That costs welding fuel though...

the fuel isn't adding any material to it though.
granted, i kinda do agree with shards being less, since they're.. not refined or whatever.

**Comment by Fordoxia** [comment]
> > Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).
> 
> That costs welding fuel though...
> 
> > now that jani borgs can refill their replacers with glass, you should remove the free lights from charges :)
> 
> Seconded

Out-of-scope for this PR probably. That said I absolutely support it.

**Comment by 1080pCat** [comment]
> > > Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).
> > 
> > 
> > That costs welding fuel though...
> > > now that jani borgs can refill their replacers with glass, you should remove the free lights from charges :)
> > 
> > 
> > Seconded
> 
> Out-of-scope for this PR probably. That said I absolutely support it.

i wouldn't quite call it outa pr scope since you're already changing values around, and generally rebalancing how/when borgs can get new lights.
im not the pr author though

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 2 above)
Comment (Fordoxia) [reply code/game/objects/items/stacks/sheets/glass.dm:91]: Mistake.

File: code/game/objects/items/devices/lightreplacer.dm (hunk 3 above)
Comment (Fordoxia) [reply code/game/objects/items/devices/lightreplacer.dm]: It has been refactored again, you immediately return if you are at or above the limit.

**Comment by DGamerL** [comment]
> i wouldn't quite call it outa pr scope since you're already changing values around, and generally rebalancing how/when borgs can get new lights.
im not the pr author though

Out of scope. We try to keep PRs as atomic as possible and encourage PR authors to split up their PRs if needed!

**Comment by Adrer** [comment]
> > Shards now refill 1 full light rather than 0.75 of a light (it's literally the same amount of material, a welding tool can turn it into a sheet).
> 
> That costs welding fuel though...

A shard can also just be put into the ORM/Autolathe to get a full sheet out of it

File: code/game/objects/items/devices/lightreplacer.dm [hunk 6]
```diff
@@ -68,45 +68,47 @@
...
+			to_chat(user, "<span class='warning'>[L] is stuck to your hand!</span>")
+			return
```
[the remark this answers is not in the document]
Comment (Fordoxia) [reply code/game/objects/items/devices/lightreplacer.dm:108]: True... Done.

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/items/devices/lightreplacer.dm [hunk 7]
```diff
@@ -68,45 +68,45 @@
...
-		if(uses >= max_uses)
-			to_chat(user, "<span class='warning'>[src] is full.</span>")
-			return
-		else if(G.use(decrement))
+
+		if(G.use(decrement))
 			AddUses(increment)
-			to_chat(user, "<span class='notice'>You insert a piece of glass into [src]. You have [uses] light\s remaining.</span>")
-			return
+			to_chat(user, "<span class='notice'>You insert a some glass into [src]. You have [uses] light\s remaining.</span>")
```
Comment (DGamerL) [inline code/game/objects/items/devices/lightreplacer.dm:81]:
```suggestion
			to_chat(user, "<span class='notice'>You insert some glass into [src]. You have [uses] light\s remaining.</span>")
```
Italian light replacer?

File: code/game/objects/items/devices/lightreplacer.dm [hunk 8]
```diff
@@ -228,14 +228,29 @@
...
+			attackby(target, U)
+			return
+
+		if(istype(target, /obj/item/light))
+			attackby(target, U)
+			return
+
+		if(istype(target, /obj/item/stack/sheet/glass))
+			attackby(target, U)
+			return
```
Comment (DGamerL) [inline code/game/objects/items/devices/lightreplacer.dm:244]:
```suggestion
		attackby(target, U)
		return
```
Why not `attackby()` anyways as that already checks for type?

**Review by DGamerL — APPROVED** [review]

**Pull request merged**